test/sp_constexpr_test2.cpp

```diff
@@ -65,4 +66,23 @@ int main()
     return boost::report_errors();
 }
 
+#ifdef __cpp_constexpr_dynamic_alloc
+struct dummy {
+  // no-ops, so safe on pointers to static constexpr variables
+  friend constexpr void intrusive_ptr_add_ref(const dummy *) {}
+  friend constexpr void intrusive_ptr_release(const dummy *) {}
```

Comment:
For my use case I only care about being able to create null intrusive_ptr objects in constant expressions. But I could  imagine someone wanting it to work with a non-atomic ref count as well. This seems sufficient to test that intrusive_ptr would support that use case without actually building that thing.